cmake/onnxruntime_util.cmake

```diff
@@ -15,6 +15,11 @@ target_include_directories(onnxruntime_util PRIVATE ${ONNXRUNTIME_ROOT} PUBLIC $
 if (onnxruntime_USE_CUDA)
  target_include_directories(onnxruntime_util PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 endif()
+if (onnxruntime_ENABLE_TRAINING)
+  # DLPack is a header-only dependency
+  set(DLPACK_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/external/dlpack/include)
+  target_include_directories(onnxruntime_util PRIVATE ${ORTTRAINING_ROOT} ${DLPACK_INCLUDE_DIR})
+endif()
```

Comment:
FYI, Vincent also moved this depedency to core. 

His PR will be merged today.

Reply:
Yeah, we synced yesterday. Going to do a  merge once his change is in.

Reply:
will defer this sync a bit later, I am putting more focus on the memory issue for now

Reply:
done.